### PR TITLE
Port plan_execution to ROS2

### PR DESCRIPTION
**Pull request by JafarAbdi**

### CODE REVIEW

File: moveit_ros/planning/plan_execution/CMakeLists.txt [hunk 1]
```diff
@@ -4,12 +4,25 @@ add_library(${MOVEIT_LIB_NAME}
...
   moveit_trajectory_execution_manager
+  moveit_core
+  moveit_planning_scene
+  moveit_planning_scene_monitor
+  rclcpp
+  Boost
+  class_loader
+  pluginlib)
+
+target_link_libraries(${MOVEIT_LIB_NAME}
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/CMakeLists.txt]: Please remove, this is redundant with `ament_target_dependencies`

File: moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_execution.h [hunk 2]
```diff
@@ -90,7 +90,8 @@ class PlanExecution
   };
 
   PlanExecution(const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor,
-                const trajectory_execution_manager::TrajectoryExecutionManagerPtr& trajectory_execution);
+                const trajectory_execution_manager::TrajectoryExecutionManagerPtr& trajectory_execution,
+                const std::shared_ptr<rclcpp::Node> node);
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_execution.h]: Move node to first position to keep it consistent with the other classes. Also make it const reference.

File: moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_with_sensing.h [hunk 3]
```diff
@@ -54,7 +54,8 @@ MOVEIT_CLASS_FORWARD(PlanWithSensing)
 class PlanWithSensing
 {
 public:
-  PlanWithSensing(const trajectory_execution_manager::TrajectoryExecutionManagerPtr& trajectory_execution);
+  PlanWithSensing(const trajectory_execution_manager::TrajectoryExecutionManagerPtr& trajectory_execution,
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_with_sensing.h]: Same

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 4]
```diff
@@ -173,7 +173,7 @@ void plan_execution::PlanExecution::planAndExecuteHelper(ExecutableMotionPlan& p
   do
   {
     replan_attempts++;
-    ROS_INFO_NAMED("plan_execution", "Planning attempt %u of at most %u", replan_attempts, max_replan_attempts);
+    RCLCPP_INFO(node_->get_logger(), "Planning attempt %u of at most %u", replan_attempts, max_replan_attempts);
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: Add global `LOGGER`

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 5]
```diff
@@ -200,8 +200,9 @@ void plan_execution::PlanExecution::planAndExecuteHelper(ExecutableMotionPlan& p
       if (plan.error_code_.val == moveit_msgs::msg::MoveItErrorCodes::UNABLE_TO_AQUIRE_SENSOR_DATA &&
           opt.replan_delay_ > 0.0)
       {
-        ros::WallDuration d(opt.replan_delay_);
-        d.sleep();
+        auto d = std::chrono::duration<double>(opt.replan_delay_);
+        auto period = std::chrono::milliseconds(int(d.count()*1000));
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: This is overly complicated, also `sleep_for` expects nanoseconds. Just cast `d` (better rename to replan_delay_seconds or so) directly to nanoseconds.

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 6]
```diff
@@ -236,29 +237,32 @@ void plan_execution::PlanExecution::planAndExecuteHelper(ExecutableMotionPlan& p
       // othewrise, we wait (if needed)
       if (opt.replan_delay_ > 0.0)
       {
-        ROS_INFO_NAMED("plan_execution", "Waiting for a %lf seconds before attempting a new plan ...",
+        RCLCPP_INFO(node_->get_logger(), "Waiting for a %lf seconds before attempting a new plan ...",
                        opt.replan_delay_);
-        ros::WallDuration d(opt.replan_delay_);
-        d.sleep();
-        ROS_INFO_NAMED("plan_execution", "Done waiting");
+        auto d = std::chrono::duration<double>(opt.replan_delay_);
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: same

File: moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp [hunk 7]
```diff
@@ -129,10 +131,9 @@ void plan_execution::PlanWithSensing::displayCostSources(bool flag)
 {
   if (flag && !display_cost_sources_)
     // publisher for cost sources
-    cost_sources_publisher_ =
-        node_handle_.advertise<visualization_msgs::MarkerArray>("display_cost_sources", 100, true);
+    cost_sources_publisher_ = node_->create_publisher<visualization_msgs::msg::MarkerArray>("display_cost_sources",rmw_qos_profile_default);
   else if (!flag && display_cost_sources_)
-    cost_sources_publisher_.shutdown();
+    cost_sources_publisher_.reset();
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp]: Does this really work? I could be wrong but I thought the publishers are maintained by the node and stay alive as long as the node does.

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 8]
```diff
@@ -46,6 +46,9 @@
 
 namespace plan_execution
 {
+
+rclcpp::Logger LOGGERN_PLAN_EXECUTION = rclcpp::get_logger("plan_execution");
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: `LOGGERN_*`->`LOGGER_*`

File: moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp [hunk 9]
```diff
@@ -46,6 +46,7 @@
 namespace plan_execution
 {
 // using namespace moveit_ros_planning;
+rclcpp::Logger LOGGERN_PLAN_EXECUTION = rclcpp::get_logger("plan_execution");
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp]: same

File: moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp (hunk 7 above)
Comment (JafarAbdi) [reply moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp]: I'm not sure, does this help .? [link](https://answers.ros.org/question/280108/ros2rclcpp-threadsafety-of-nodecreate_publisher/)

File: moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp [hunk 10]
```diff
@@ -258,13 +276,13 @@ bool plan_execution::PlanWithSensing::lookAt(const std::set<collision_detection:
 
   std::vector<std::string> names;
   sensor_manager_->getSensorsList(names);
-  geometry_msgs::PointStamped point;
+  geometry_msgs::msg::PointStamped point;
```
Comment (henningkayser) [inline moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp]: Use `node_->now()`

**Pull request merged**